### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by GregJMitchell**
### Description
Implements Google OAuth
Adds current_user to app
Adds authorization to controllers
Updates tests to stub current user
### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [x] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [x] I ran rubocop and fixed issues
- [x] I ran full test suite - all tests passing

### CODE REVIEW

**Review by araspaas — APPROVED** [review]

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class SessionsController < ApplicationController
+  def create
+    body = {
+      client_id: ENV['GOOGLE_CLIENT_ID'],
+      client_secret: ENV['GOOGLE_CLIENT_SECRET'],
```
Comment (helloeduardo) [inline app/controllers/sessions_controller.rb:5]: I think Phil mentioned he would take these out as they are unnecessary in our morning oauth session

File: config/initializers/omniauth.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Rails.application.config.middleware.use OmniAuth::Builder do
+  provider :google_oauth2, ENV['GOOGLE_CLIENT_ID'], ENV['GOOGLE_CLIENT_SECRET']
+end
```
Comment (helloeduardo) [inline config/initializers/omniauth.rb:3]: Thanks for working out this oauth! 🥇

File: spec/features/dashboard_spec.rb [hunk 3]
```diff
@@ -33,7 +48,27 @@
       end
     end
 
-    xit 'If I am not logged in, I am taken to the homepage' do
+    it 'If I am not logged in, I am taken to the homepage' do
```
Comment (helloeduardo) [inline spec/features/dashboard_spec.rb:51]: Great sad path testing here to redirect to the home

**Review by helloeduardo — APPROVED** [review]
Thanks for all of this work! So excited to have oauth working :)

**Review by araspaas — CHANGES_REQUESTED** [review]
We are working on this at the moment do not merge yet. we are making changes based on multiple API calls vs just the one big one

**Review by araspaas — APPROVED** [review]
Changes have been made can be merged

**Pull request merged**